flocx_market/manager/service.py

```diff
@@ -0,0 +1,43 @@
+from oslo_context import context as ctx
+from oslo_log import log as logging
+from oslo_service import service
+from oslo_service import periodic_task
+from oslo_service import threadgroup
+
+import flocx_market.db.sqlalchemy.api as dbapi
+
+import flocx_market.conf
+
+
+CONF = flocx_market.conf.CONF
+EVENT_INTERVAL = 10
+LOG = logging.getLogger(__name__)
+
+
+class ManagerService(service.Service):
+    def __init__(self, threads=1000):
+        self.tg = threadgroup.ThreadGroup(threads)
+        LOG.info("Creating flocx-market manager service")
+        self.tasks = Manager(CONF)
+        self._context = ctx.RequestContext(
+            auth_token=None,
+            project_id=None,
+            is_admin=True,
+            overwrite=False)
+
+    def start(self):
+        LOG.info("Starting flocx-market manager service")
+        self.tg.add_dynamic_timer(
+            self.tasks.run_periodic_tasks,
+            initial_delay=None,
+            periodic_interval_max=1,
+            context=ctx
+        )
+
+
+class Manager(periodic_task.PeriodicTasks):
+
+    @periodic_task.periodic_task(spacing=1, run_immediately=True)
+    def bar(self, context):
+        all_offers = dbapi.offer_get_all()
+        return [x.to_dict() for x in all_offers]
```

Comment:
I think you're actually really close to making this a real piece of functionality! Here's what I'd suggest:

* Update dbapi to include a query for `offer_get_all_by_status` - or maybe specifically `offer_get_all_unexpired`, now that I think about it
* Use that query here
* check each offer's end_time against the current time, and if end_time is greater, then set the status to `expired`